### ISSUE TITLE
Questions about the experiment setup

### ISSUE THREAD

**Issue by hollow1024** [ExplainableML/TCAF-GZSL#9]
Hello! I encountered the following two problems: 1. The training phase in the experiment is divided into two stages, so what is the purpose of setting the second stage, why can the HM value of the second stage reach 1, if the best_epoch of the first stage is very small, such as 1, then the HM value of the second stage cannot reach 1, will this have an impact on the future evaluation results?2. In data reading x_p_a dimension with x_p_v is 3, so what is the meaning of the values of the second dimension (15 and 23)? How did you get it?

**Comment by hummelth** [comment]
Hi! 
1. As mentioned in the paper (end of page 9), we train all models with a two-stage training protocol. In the first stage, we determine the optimal parameters for the calibrated stacking and the amount of required training epochs. This is to ensure that we do not optimize these on the test set. In the second stage, we then train on the training and validation set from the first stage using the parameters from Stage 1. In the code for the second stage, we still evaluate the HM on the validation, but, as we also train on it, the reported validation numbers do not reflect the true performance. Afterward, we test the model on the test set.
2. The numbers represent the length of the temporal dimension for the audio and video data. Due to their different feature extraction rates (0.64 seconds for C3D and 0.96 seconds for VGGish), both modalities have different temporal lengths.

I hope this clarifies!

**Comment by hollow1024** [comment]
Thanks for the answer！
